docs/en/04_Changelogs/4.0.0.md

```diff
@@ -1,197 +1,107 @@
+* CMS CSS has been re-developed using bootstrap 4 as a base.
+* Asset admin has been replaced with a purely ReactJS powered upgrade.
+* Versioning is now a much more powerful feature, with the addition of campaigns to allow batches of related
+  or inter-dependent objects to be published as a single act. Dependencies between versioned objects can be
+  declared using the new ownership API, so that developers can ensure that relational consistency is
+  maintained during publishing. This new system can be managed via the new "Campaigns" CMS section.
+* Template variable casting (e.g. `<h1>$Title</h1>`) is enforced by default, which will ensure safe HTML encode
+  unless explicitly opted out.
+* Themes are now configured to cascade, where you can specify a list of themes, and have the template engine
+  search programatically through a prioritised list when resolving template and CSS file paths.
+
+## <a name="upgrading"></a>Upgrading
+
+The below sections describe how to go about updating an existing site to be prepared for upgrade to 4.0.
+Most of these upgrading tasks will involve manual code review, although in some cases there are
+some automated processes that users can run to 
+
+### <a name="upgrading-primary"></a>Standard Upgrade
+
+This section describes the processes which every project upgrading to 4.0 should follow. This should be followed
+as a standard first point of upgrade.
+
+#### Update references to renamed and namespaced classes
+
+4.0 moves many classes to namespaces, and renames many methods.
+A tool is available to (semi-)automatically update your 3.x code to the new 4.0 names.
+
+See https://github.com/silverstripe/silverstripe-upgrader/ for the latest documentation on 
+how to upgrade your code. Typically the process will be:
+
+	
+	composer global require silverstripe/upgrader
+	cd ~/Project/Root
+	~/.composer/vendor/bin/upgrade-code upgrade ./mysite --write
+
+
+If you want to do a dry-run, omit the `--write` option to see a preview of a diff of
+all changed project files.
+
+This will resolve the majority of upgrading work, but for specific changes that will
+require manual intervention, please see the below upgrading notes.
+
+#### Update tempalate locations and references
+
+Templates are now much more strict about their locations. You can no longer put a template in an arbitrary
+folder and have it be found. Case is now also checked on case-sensitive filesystems.
+
+Either include the folder in the template name (`renderWith('MyEmail.ss')` => `renderWith('emails/MyEmail.ss')`),
+move the template into the correct directory, or both.
+
+When using `<% include %>` template tag you should continue to leave out the `Includes` folder. 
+`<% include Sidebar %>` will match only match `Includes/Sidebar.ss`, not `Sidebar.ss`.
+Please refer to our [template syntax](/developer_guides/templates/syntax) for details.
+
+The `forms` template folder is no longer used to lookup templates for `FormField` instances.
```

Comment:
I'd generalise this to "Core template locations have moved - if you're including or overriding these (e.g. for `FormField` templates) please adjust to the new paths"